fix and unify building of local package cache

lethliel opened the pull request:
* all filename functions now return bytes-like objects
* the caller does the decoding
* the caller in build.py passes encoded arguments

fixes https://github.com/openSUSE/osc/issues/613

lethliel commented:
@marcus-h: please review

marcus-h commented:
On 2019-07-26 01:50:12 -0700, Marco Strigl wrote:
> * the caller does the decoding

(From a conceptual POV, the callers (like the one in move_package
(fetch.py)) should not decode the result.)

> diff --git a/osc/util/rpmquery.py b/osc/util/rpmquery.py
> index 535cd37e..07a62b6b 100644
> --- a/osc/util/rpmquery.py
> +++ b/osc/util/rpmquery.py
> @@ -374,7 +374,7 @@ def rpmvercmp(ver1, ver2):
>
>      @staticmethod
>      def filename(name, epoch, version, release, arch):
> -        return '%s-%s-%s.%s.rpm' % (name, version, release, arch)
> +        return b'%s-%s-%s.%s.rpm' % (name, version, release, arch)

We have to adjust the canonname() method as well (the hardcoded arch
strs should be bytes).

Otherwise, LGTM!:)

lethliel commented:
On 7/26/19 12:41 PM, Marcus Hüwe  wrote:
> On 2019-07-26 01:50:12 -0700, Marco Strigl wrote:
>> * the caller does the decoding
> 
> (From a conceptual POV, the callers (like the one in move_package
> (fetch.py)) should not decode the result.)
> 
>> diff --git a/osc/util/rpmquery.py b/osc/util/rpmquery.py
>> index 535cd37e..07a62b6b 100644
>> --- a/osc/util/rpmquery.py
>> +++ b/osc/util/rpmquery.py
>> @@ -374,7 +374,7 @@ def rpmvercmp(ver1, ver2):
>>
>> @staticmethod
>> def filename(name, epoch, version, release, arch):
>> - return '%s-%s-%s.%s.rpm' % (name, version, release, arch)
>> + return b'%s-%s-%s.%s.rpm' % (name, version, release, arch)
> 
> We have to adjust the canonname() method as well (the hardcoded arch
> strs should be bytes).

Yes. You are right. Changed that now.

> 
> Otherwise, LGTM!:)
>

marcus-h commented:
On 2019-07-26 04:12:30 -0700, Marco Strigl wrote:
> On 7/26/19 12:41 PM, Marcus Hüwe  wrote:
> > On 2019-07-26 01:50:12 -0700, Marco Strigl wrote:
> >> * the caller does the decoding
> >
> > (From a conceptual POV, the callers (like the one in move_package
> > (fetch.py)) should not decode the result.)
> >
> >> diff --git a/osc/util/rpmquery.py b/osc/util/rpmquery.py
> >> index 535cd37e..07a62b6b 100644
> >> --- a/osc/util/rpmquery.py
> >> +++ b/osc/util/rpmquery.py
> >> @@ -374,7 +374,7 @@ def rpmvercmp(ver1, ver2):
> >>
> >> @staticmethod
> >> def filename(name, epoch, version, release, arch):
> >> - return '%s-%s-%s.%s.rpm' % (name, version, release, arch)
> >> + return b'%s-%s-%s.%s.rpm' % (name, version, release, arch)
> >
> > We have to adjust the canonname() method as well (the hardcoded arch
> > strs should be bytes).
>
> Yes. You are right. Changed that now.
>
Thanks! Can you also adjust RepoDataQueryResult.canonname, too?

lethliel commented:
On 7/26/19 1:17 PM, Marcus Hüwe  wrote:
> On 2019-07-26 04:12:30 -0700, Marco Strigl wrote:
>> On 7/26/19 12:41 PM, Marcus Hüwe wrote:
>> > On 2019-07-26 01:50:12 -0700, Marco Strigl wrote:
>> >> * the caller does the decoding

[SNIP]

 @staticmethod
>> >> def filename(name, epoch, version, release, arch):
>> >> - return '%s-%s-%s.%s.rpm' % (name, version, release, arch)
>> >> + return b'%s-%s-%s.%s.rpm' % (name, version, release, arch)
>> >
>> > We have to adjust the canonname() method as well (the hardcoded arch
>> > strs should be bytes).
>>
>> Yes. You are right. Changed that now.
>>
> Thanks! Can you also adjust RepoDataQueryResult.canonname, too?
> 

adjusted RepoDataQueryResult.canonname but I do not see where this is ever used.

Thanks and Cheers,
Marco

marcus-h commented:
On 2019-07-26 04:43:34 -0700, Marco Strigl wrote:
> On 7/26/19 1:17 PM, Marcus Hüwe  wrote:
> > On 2019-07-26 04:12:30 -0700, Marco Strigl wrote:
> >> On 7/26/19 12:41 PM, Marcus Hüwe wrote:
> >> > On 2019-07-26 01:50:12 -0700, Marco Strigl wrote:
> >> >> * the caller does the decoding
>
> [SNIP]
>
>  @staticmethod
> >> >> def filename(name, epoch, version, release, arch):
> >> >> - return '%s-%s-%s.%s.rpm' % (name, version, release, arch)
> >> >> + return b'%s-%s-%s.%s.rpm' % (name, version, release, arch)
> >> >
> >> > We have to adjust the canonname() method as well (the hardcoded arch
> >> > strs should be bytes).
> >>
> >> Yes. You are right. Changed that now.
> >>
> > Thanks! Can you also adjust RepoDataQueryResult.canonname, too?
> >
>
> adjusted RepoDataQueryResult.canonname but I do not see where this is ever used.
>
It is currently unused (there could be 3rd-party callers, though).
Merged. Thanks!
